macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,445 @@
+        enum: init_option
+      init_comp_mask:
+        -orig-id: initCompMask
+        value: 0xFF00
+        doc: mask for compression IDs
+      comp_init:
+        value: options & init_comp_mask
+        enum: init_option
+  sound_command:
+    seq:
+      - id: is_data_offset
+        type: b1
+      - id: cmd
+        type: b15
+        enum: cmd_type
+      - id: param1
+        type: u2
+      - id: param2
+        type: u4
+    instances:
+      sound_header:
+        pos: param2
+        type: sound_header
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+  sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_samples
+        -orig-id: length
+        type: u4
+        doc: number of samples
+        if: sound_header_type == sound_header_type::standard
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: midi_note
+        type: u1
+        doc: base frequency of sample, expressed as MIDI note values, 60 is middle C
+        -orig-id: baseFrequency
+      - id: extended_or_compressed
+        type: extended_or_compressed
+        if: sound_header_type == sound_header_type::extended or sound_header_type == sound_header_type::compressed
+      - id: sample_area
+        -orig-id: sampleArea
+        size: |
+          sound_header_type == sound_header_type::standard ? num_samples :
+          sound_header_type == sound_header_type::extended ? extended_or_compressed.num_frames * num_channels * extended_or_compressed.bits_per_sample / 8 :
+          _io.size - _io.pos
+        doc: sampled-sound data
+        if: sample_ptr == 0
+    instances:
+      base_freqeuncy:
+        value: _root.midi_note_to_frequency[midi_note]
+        #TODO: If https://github.com/kaitai-io/kaitai_struct/issues/216 is implemented:
+        #TODO: value: (2 ** ((midi_note - 69) / 12)) * 440
+        doc: |
+          base frequency of sample in Hz
+          Calculated with the formula (2 ** ((midi_note - 69) / 12)) * 440
+        doc-ref: https://en.wikipedia.org/wiki/MIDI_tuning_standard
+        if: midi_note >= 0 and midi_note < 128
+      sound_header_type:
+        pos: 20
+        type: u1
+        enum: sound_header_type
```

Comment:
Oh, sorry, I forgot to add to https://github.com/kaitai-io/kaitai_struct_formats/pull/299#discussion_r506129877 that this won't work anymore. You need to offset the `pos: 20` relatively to the start of the `sound_header`. A common solution is to capture and save the `_io.pos` in the beginning of the `sound_header`, like this:

```ksy
types:
  sound_header:
    seq:
      - size: 0
        if: start_ofs < 0 # invoking the `start_ofs` value instance to save the current `_io.pos`
      - id: sample_ptr
        -orig-id: samplePtr
        type: u4
      # ...
    instances:
      start_ofs:
        value: _io.pos
      sound_header_type:
        pos: start_ofs + 20
        type: u1
        enum: sound_header_type
```

Or, you can also move the `sound_header_type` outside of the `sound_header` like you had before:

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L70

https://github.com/kaitai-io/kaitai_struct_formats/blob/9adbc8f49d6000f0023dd8c1b8ed2d3faae37d16/macos/mac_os_resource_snd.ksy#L81-L86

However, maybe I prefer the version with capturing `start_ofs`, because that's basically guaranteed to work under all circumstances, regardless of whether the `sound_header` is wrapped in a substream or not. With the `pos: param2 + 20` version, there's a room for errors like `param1 + 20` etc.